is_core/generic_views/form_views.py

```diff
@@ -308,54 +308,54 @@ def generate_readonly_fields(self):
     def get_inline_views(self):
         return self.inline_views or ()
 
+    def generate_inline_views(self):
+        inline_views = self.get_inline_views()
+        fieldsets = self.get_fieldsets()
+        if inline_views and fieldsets:
+            raise ImproperlyConfigured('You can define only one of inline views and fieldsets')
+        return inline_views or get_inline_views_from_fieldsets(fieldsets)
+
     def init_inline_views(self, instance):
-        inline_views = OrderedDict()
-        for inline_view in self.get_inline_views():
-            inline_views[inline_view.__name__] = inline_view(self.request, self, instance)
-        return inline_views
+        return [inline_view(self.request, self, instance) if isinstance(inline_view, type) else inline_view
+                for inline_view in self.generate_inline_views()]
 
     def _filter_inline_form_views(self, inline_views):
-        inline_form_views = OrderedDict()
-        for name, view in inline_views.items():
-            if isinstance(view, InlineFormView):
-                inline_form_views[name] = view
-        return inline_form_views
+        return [view for view in inline_views if isinstance(view, InlineFormView)]
 
-    def generate_fieldsets(self):
-        fieldsets = self.get_fieldsets()
+    def generate_fieldsets(self, inline_views=None, **kwargs):
+        fieldsets = deepcopy(self.get_fieldsets())
 
         if fieldsets is not None:
+            inline_view_opts = get_inline_views_opts_from_fieldsets(fieldsets)
+            for inline_view_inst, inline_view_opt in zip(inline_views, inline_view_opts):
+                inline_view_opt['inline_view'] = inline_view_inst
             return fieldsets
         else:
             fieldsets = [(None, {'fields': self.generate_fields() or
                                  list(self.generate_form_class().base_fields.keys())})]
-            for inline_view in self.get_inline_views():
-                if (issubclass(inline_view, InlineFormView) and (not inline_view.max_num or
+            for inline_view in inline_views:
+                if (isinstance(inline_view, InlineFormView) and (not inline_view.max_num or
                                                                  inline_view.max_num > 1)):
                     title = inline_view.model._meta.verbose_name_plural
                 else:
                     title = inline_view.model._meta.verbose_name
-                fieldsets.append((title,
-                                  {'inline_view': inline_view.__name__}))
+                fieldsets.append((title, {'inline_view': inline_view}))
```

Comment:
tady se generuje tedy deepcopy se tu řešit nemusí rovnou tam dám instanci, tato metoda se volá jen jednou v get_context_data (pro úplnost)